### PR TITLE
API revision

### PR DESCRIPTION
**Pull request by scottinet**
# Description

Koncorde's API has a few oddities, remnants of its early stages, when the engine was a part of Kuzzle.

This includes:
* index and collection (required arguments) have been replaced by an `index` option: index/collection made Koncorde harder to use for no obvious reason. Instead, Koncorde now manages a default index and, as an option, named indexes can be used to isolate filters. These are filter indexes, they have nothing to do with Kuzzle's indexes.
* the `ids` keyword has no value for Koncorde and has been removed from its documentation. It's just a shortcut to the `equals` keyword, at the price of a clunky API: an optional "id" argument in the `test` function for no obvious reason other than "if you want that keyword to be used, you need to pass an additional argument". I left the keyword in the code though, to prevent breaking changes in Kuzzle. That way, adapting Kuzzle to Koncorde v4 will be easy: just inject the `_id` value in the data passed to Koncorde beforehand, and voila. 

# Other changes

* The Koncorde.register and Koncorde.store functions have been simplified: they now both return the filter unique identifier directly, instead of an object
* Performances: oopsie... years ago I forgot to convert the normalized filter dynamic object to a plain class (optimizable to a C++ class by V8). That object is only stored and used everywhere by Koncorde, so yeah, there is a noticeable performances improvement there
* Fix a regexp issue raised by LGTM (potentially dangerous regexp used when parsing geographical coordinates)

### CODE REVIEW

**Comment by scottinet** [comment]
About the `id` keyword being removed, I wonder... Maybe I should put it back as a shortcut to `equals: { _id: 'xxx' }`, to prevent a breaking change in Kuzzle, but with the following differences:

* leave it undocumented: this keyword should really not be used anymore
* let Kuzzle inject the `_id` field in data to be tested by itself: I don't think that having the `id` option in the test method is beneficial for Koncorde

wdyt?

File: lib/transform/standardize.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ class Standardizer {
     }
 
     if (!this.allowedKeywords.has(keywords[0])) {
-      throw new Error(`Unknown DSL keyword: ${keywords[0]}`);
+      throw new Error(`Unknown Koncorde keyword: ${keywords[0]}`);
```
Comment (Aschen) [inline lib/transform/standardize.js:81]: Can we have a custom error with the `keyword` property so we can construct higher level error messages in project using Koncorde?

**Review by Aschen — CHANGES_REQUESTED** [review]
I'm not fond of the `index` terminology because it's still bound to Kuzzle to me. 
I would rather have something more neutral like `scope`

File: lib/transform/standardize.js (hunk 1 above)
Comment (scottinet) [reply lib/transform/standardize.js:81]: yes that will come soon

**Review by Aschen — APPROVED** [review]

**Review by Shiranuit — APPROVED** [review]

**Pull request merged**